src/main.rs

```diff
@@ -1,3 +1,88 @@
+use git2::{Commit, Error, Time};
+use git_graph::graph::GitGraph;
+
+struct Args {}
+
 fn main() {
-    println!("Hello, world!");
+    let args = Args {};
+    match run(&args) {
+        Ok(()) => {}
+        Err(e) => println!("error: {}", e),
+    }
+}
+
+fn run(_args: &Args) -> Result<(), Error> {
+    let path = ".";
+    let graph = GitGraph::new(path)?;
+    for info in &graph.commits {
+        let commit = &graph.commit(info.oid)?;
+        print_commit_short(commit, &info.branches);
+    }
+    Ok(())
+}
+
+#[allow(dead_code)]
```

Comment:
I don't think this one is dead? (Not a friend of committed dead generally to be honest.)

Reply:
Removed dead code, or the attribute where not valid anymore.